incubator/hnc/pkg/controllers/object_controller_test.go

```diff
@@ -100,6 +99,15 @@ var _ = Describe("Secret", func() {
 		Eventually(hasSecret(ctx, bazName, "foo-sec")).Should(BeFalse())
 	})
 
+	It("should not be propagated if it has a condition", func() {
+
+		setParent(ctx, barName, fooName)
+		setParent(ctx, quxName, barName)
+		if hasAnyCondition(ctx, barName) {
+			panic("CONDITION SET")
+		}
```

Comment:
Based on https://onsi.github.io/gomega/#making-assertions, I'd say something like:

`Expect(hasAnyCondition(ctx, barName)).ToNot(Equal(true))`

You shouldn't panic since it stops the rest of the tests from running.

Usually, you'll use `Eventually()` instead of `Expect()` since the latter runs immediately while the former can retry for up to two seconds (configurable in `suite_test.go` IIRC, but please don't change it). This is necessary because the controller is running asynchronously alongside the test code, so when you make a change (like updating the test apiserver), it takes a few moments to respond.